_pages/documentation/general_docs/developement/coding_style.md

```diff
@@ -0,0 +1,378 @@
+
+Lines must be a maximum of 79 characters long. 
+
+## Braces
+
+For control blocks (if, while, etc.), opening braces must be on the same line as the control keyword with a space between the closing parenthesis and the opening brace.
+
+* Exception: for multi-line expressions, the opening brace may be placed on a separate line to distinguish the control block from the statements inside the block.
+```c++
+if (...) {
+    ...
+}
+
+// exception case
+for (...;
+     ...;
+     ...) // brace could be up here
+{ // but this is optionally OK *only* when the 'for' spans multiple lines
+    ...
+}
+```
+'Else' keywords should follow the closing 'if' brace on the same line, as follows:
+```c++
+if (...) {
+    ...
+} else if (...) {
+    ...
+} else {
+    ...
+}
+```
+Blocks that consist of a single statement that fits on a single line may optionally omit the braces. Braces are still required if the single statement spans multiple lines, or if the block is part of an else/if chain where other blocks have braces.
+
+```c++
+// This is OK with or without braces
+if (a > 0)
+    --a;
+
+// In the following cases, braces are still required
+if (a > 0) {
+    obnoxiously_named_function_with_lots_of_args(verbose_arg1,
+                                                 verbose_arg2,
+                                                 verbose_arg3);
+}
+
+if (a > 0) {
+    --a;
+} else {
+    underflow = true;
+    warn("underflow on a");
+}
+```
+For function definitions or class declarations, the opening brace must be in the first column of the following line.
+
+In function definitions, the return type should be on one line, followed by the function name, left-justified, on the next line. As mentioned above, the opening brace should also be on a separate line following the function name.
+
+See examples below:
+```c++
+int
+exampleFunc(...)
+{
+    ...
+}
+
+class ExampleClass
+{
+  public:
+    ...
+};
+```
+
+Functions should be preceded by a block comment describing the function.
+
+Inline function declarations longer than one line should not be placed inside class declarations. Most functions longer than one line should not be inline anyway. 
+
+## Spacing
+
+There should be:
+
+* one space between keywords (if, for, while, etc.) and opening parentheses
+* one space around binary operators (+, -, <, >, etc.) including assignment operators (=, +=, etc.)
+* no space around '=' when used in parameter/argument lists, either to bind default parameter values (in Python or C++) or to bind keyword arguments (in Python)
+* no space between function names and opening parentheses for arguments
+* no space immediately inside parentheses, except for very complex expressions. Complex expressions are preferentially broken into multiple simpler expressions using temporary variables.
+
+
+For pointer and reference argument declarations, either of the following are acceptable:
+
+    FooType *fooPtr;
```

Comment:
I think it's better to wrap in ``` and call it out that's it's c++

Like above. I can't show you what I mean because github markdown keeps rendering what I'm trying to show ;)

Reply:
Yes, understood, I will change it, thanks.